cmd/soroban-rpc/internal/db/event.go

```diff
@@ -97,3 +109,169 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 
 	return nil
 }
+
+type ScanFunction func(xdr.DiagnosticEvent, events.Cursor, int64, *xdr.Hash) bool
+
+// trimEvents removes all Events which fall outside the ledger retention window.
+func (eventHandler *eventHandler) trimEvents(latestLedgerSeq uint32, retentionWindow uint32) error {
```

Comment:
I see this function is the exact copy of `trimTransactions` function. Would it be a good idea to just combine and use a common function and pass the event/txn table related info as params?

Reply:
I am not planning to do this refactor in this PR to reduce the chaos :)  This is a good idea and added the issue [here](https://github.com/stellar/soroban-rpc/issues/259)